cmd/nitro/nitro.go

```diff
@@ -830,79 +808,22 @@ func ParseNode(ctx context.Context, args []string) (*NodeConfig, *genericconf.Wa
-		"l2.chain-id":                                            42170,
-		"init.empty":                                             true,
-	}, "."), nil)
-}
-
-func applyArbitrumRollupGoerliTestnetParameters(k *koanf.Koanf) error {
-	return k.Load(confmap.Provider(map[string]interface{}{
-		"persistent.chain":                   "goerli-rollup",
-		"node.forwarding-target":             "https://goerli-rollup.arbitrum.io/rpc",
-		"node.feed.input.url":                "wss://goerli-rollup.arbitrum.io/feed",
-		"l1.rollup.bridge":                   "0xaf4159a80b6cc41ed517db1c453d1ef5c2e4db72",
-		"l1.rollup.inbox":                    "0x6bebc4925716945d46f0ec336d5c2564f419682c",
-		"l1.rollup.rollup":                   "0x45e5caea8768f42b385a366d3551ad1e0cbfab17",
-		"l1.rollup.sequencer-inbox":          "0x0484a87b144745a2e5b7c359552119b6ea2917a9",
-		"l1.rollup.validator-utils":          "0x344f651fe566a02db939c8657427deb5524ea78e",
-		"l1.rollup.validator-wallet-creator": "0x53eb4f4524b3b9646d41743054230d3f425397b3",
-		"l1.rollup.deployed-at":              7217526,
-		"l2.chain-id":                        421613,
-		"init.empty":                         true,
-	}, "."), nil)
-}
-
-func applyArbitrumRollupRinkebyTestnetParameters(k *koanf.Koanf) error {
-	return k.Load(confmap.Provider(map[string]interface{}{
-		"persistent.chain":                   "rinkeby-nitro",
-		"node.forwarding-target":             "https://rinkeby.arbitrum.io/rpc",
-		"node.feed.input.url":                "wss://rinkeby.arbitrum.io/feed",
-		"l1.rollup.bridge":                   "0x85c720444e436e1f9407e0c3895d3fe149f41168",
-		"l1.rollup.inbox":                    "0x578BAde599406A8fE3d24Fd7f7211c0911F5B29e",
-		"l1.rollup.rollup":                   "0x71c6093c564eddcfaf03481c3f59f88849f1e644",
-		"l1.rollup.sequencer-inbox":          "0x957c9c64f7c2ce091e56af3f33ab20259096355f",
-		"l1.rollup.validator-utils":          "0x0ea7372338a589e7f0b00e463a53aa464ef04e17",
-		"l1.rollup.validator-wallet-creator": "0x237b8965cebe27108bc1d6b71575c3b070050f7a",
-		"l1.rollup.deployed-at":              11088567,
-		"l2.chain-id":                        421611,
-	}, "."), nil)
-}
-
-func applyArbitrumAnytrustGoerliTestnetParameters(k *koanf.Koanf) error {
-	return k.Load(confmap.Provider(map[string]interface{}{
-		"persistent.chain": "goerli-anytrust",
-	}, "."), nil)
+func applyChainParameters(k *koanf.Koanf, chainId uint64, l1ChainId uint64, l2ChainInfoFiles []string) (bool, error) {
+	chainInfo, err := chaininfo.ProcessChainInfo(chainId, l2ChainInfoFiles)
+	if err != nil {
+		return false, err
+	}
+	if chainInfo != nil {
+		if chainInfo.ParentChainId != l1ChainId {
+			return false, fmt.Errorf("ParentId: %d provided in %s for chainId: %d is not equal to l1ChainId: %d provided in commandline", chainInfo.ParentChainId, l2ChainInfoFiles, chainId, l1ChainId)
+		}
+		err = k.Load(rawbytes.Provider(*chainInfo.ChainParameters), json.Parser())
```

Comment:
we should check if `chainInfo.ChainParameters` is not `nil` to avoid panic

Reply:
thanks for catching this, done.